src/Lecture2.hs

```diff
@@ -157,8 +172,35 @@ data Knight = Knight
     , knightEndurance :: Int
     }
 
+-- mine code start, dragonFight still NOT IMPLEMENTED
+
+data Chest a = MkChest
+  {
+    chestGold :: Int,
+    chestTreasure :: a
+  }
+  deriving Show
+
+data DragonType = 
+  DragonRed |
+  DragonBlack |
+  DragonGreen
+  deriving Show
+
+type Experince = Int
+
+getExperienceByDragon :: DragonType -> Experince
+getExperienceByDragon dragon = case dragon of 
+  DragonRed -> 100
+  DragonBlack -> 150
+  DragonGreen -> 250
+
+-- mine code end
```

Comment:
That's a good start! 👍🏻 
Do you need any help or are you currently working on this?